### ISSUE TITLE
`but worktree new`: let the caller choose the destination path and name

### ISSUE THREAD

**Issue by artile** [gitbutlerapp/gitbutler#15208]
### Summary

`but worktree new` always creates the worktree at `<data_dir>/worktrees/<uuid-v4>`. There is no way to say where the worktree should go or what it should be called. Please accept an optional destination path and an optional name, the way `git worktree add <path>` and `jj workspace add <destination> --name <name>` already do.

### Current behaviour

In `crates/but-worktrees/src/new.rs`, `worktree_new()` generates the identifier itself and derives the path from it:

```rust
let id = WorktreeId::generate();               // uuid::Uuid::new_v4()
let path = worktree_workdir(data_dir, &id);    // data_dir/worktrees/<id>
git_worktree_add(repo.common_dir(), &path, to_checkout.detach())?;
```

`git_worktree_add()` in `crates/but-worktrees/src/git.rs` then shells out to plain `git worktree add --detach <path> <commit>`, so the underlying object is an ordinary linked worktree — only the location and the name are fixed by GitButler.

The name matters beyond looks: `WorktreeId::from_path()` in `crates/but-worktrees/src/lib.rs` takes the basename of the directory as the worktree id, and `git_worktree_remove()` passes that id straight to `git worktree remove`. So the UUID becomes the handle a user types for `but worktree destroy`, and the string shown by `but worktree list`.

### Why this is a problem

Worktrees are increasingly created by an outer tool rather than by hand: task runners and agent orchestrators cut one worktree per ticket and name it after the ticket, next to the repository, so a human, a shell and an editor can all find it. With the path and name fixed, such a tool cannot use `but worktree new` at all. It has to call `git worktree add` itself, which works — `worktree_list()` enumerates git's own registry and joins GitButler metadata by basename, so foreign worktrees do show up — but they arrive without the `gitbutler-created-from` and `gitbutler-base` sidecar files, so `created_from_ref` and `base` are empty and `worktree_integrate()` falls back to a plain merge base.

A UUID under the application data directory is also hard to work with directly: `but worktree list` prints a column of identical-looking UUIDs, and `cd` into one is a copy-paste rather than a name a person remembers.

### Proposed change

Add an optional positional destination and an optional `--name` to `but worktree new`:

```
but worktree new <branch> [<destination>] [--name <name>]
```

- `<destination>` — where to create the worktree. Relative paths resolve against the current directory, as `git worktree add` does.
- `--name` — the worktree identifier. Defaults to the basename of the destination, matching how `WorktreeId::from_path()` already derives it.
- Both omitted — today's behaviour exactly, so nothing changes for existing users.

The metadata layer needs no change: `save_worktree_meta()` writes into `$GIT_COMMON_DIR/worktrees/<name>/`, which git creates for any worktree regardless of where its working directory lives.

### Alternative considered

Documenting the two sidecar files as a public contract would also unblock external tools — they could create the worktree with git and write `gitbutler-created-from` and `gitbutler-base` themselves. That works today but relies on an internal file format, so an explicit CLI argument seems the better contract.
